src/toil/jobStores/aws/jobStore.py

```diff
@@ -301,179 +294,195 @@ def load_job(self, job_id: str):
+            encoding: Optional[str] = None,
+            errors: Optional[str] = None
+    ) -> Iterator[IO[Any]]:
+        pipe = MultiPartPipe(part_size=self.part_size,
+                             s3_resource=self.s3_resource,
                              bucket_name=self.bucket_name,
-                             file_id=f'{self.content_key_prefix}{file_id}',
-                             encryption_args=self.encryption_args)
+                             file_id=self.existing_full_s3_key_from_file_id(file_id),
+                             encryption_args=self.encryption_args,
+                             encoding=encoding,
+                             errors=errors)
         with pipe as writable:
             yield writable
 
     @contextmanager
-    def write_shared_file_stream(self, file_id, encoding=None, errors=None):
-        # TODO
-        self._requireValidSharedFileName(file_id)
-        pipe = MultiPartPipe(encoding=encoding,
-                             errors=errors,
-                             part_size=self.part_size,
-                             s3_client=self.s3_client,
+    def write_shared_file_stream(
+        self,
+        shared_file_name: str,
+        encrypted: Optional[bool] = None,
+        encoding: Optional[str] = None,
+        errors: Optional[str] = None,
+    ) -> Iterator[IO[bytes]]:
+        pipe = MultiPartPipe(part_size=self.part_size,
+                             s3_resource=self.s3_resource,
                              bucket_name=self.bucket_name,
-                             file_id=f'{self.shared_key_prefix}{file_id}',
-                             encryption_args=self.encryption_args)
+                             file_id=f'{self.shared_key_prefix}{shared_file_name}',
+                             encryption_args=self.encryption_args,
+                             encoding=encoding,
+                             errors=errors)
         with pipe as writable:
             yield writable
 
-    def update_file(self, file_id, local_path):
-        # Why use this over plain write file?
-        # TODO: job_id does nothing here without a cleanup variable
-        self.write_file(local_path=local_path, file_id=file_id)
+    def update_file(self, file_id: str, local_path: str) -> None:
+        copy_local_to_s3(
+            s3_resource=self.s3_resource,
+            local_file_path=local_path,
+            dst_bucket=self.bucket_name,
+            dst_key=self.existing_full_s3_key_from_file_id(file_id),
+            extra_args=self.encryption_args
+        )
 
-    def file_exists(self, file_id):
+    def file_exists(self, file_id: str) -> bool:
         return s3_key_exists(s3_resource=self.s3_resource,
                              bucket=self.bucket_name,
-                             key=f'{self.content_key_prefix}{file_id}',
+                             key=self.existing_full_s3_key_from_file_id(file_id),
                              extra_args=self.encryption_args)
```

Comment:
This isn't going to work because `file_exists` is meant to be able to return `False`, but `existing_full_s3_key_from_file_id` will always raise if no such key exists.